In [1]:
%matplotlib inline
%load_ext autoreload
import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt
import pandas as pd
import pathlib
import time
from datetime import datetime
import scipy.optimize as opt
np.set_printoptions(precision=3)
np.set_printoptions(suppress=True)
plt.style.use("ggplot")
mpl.pyplot.rcParams['figure.figsize'] = (5.0, 4.0)
plt.rcParams["font.size"] = 13
#mpl.rcParams["font.family"] = 'Osaka'

import sys
sys.path.append('../code/')

from simulation_additional import *
from simulation_fixed_path import make_stats
from hash import *
from plot_path import *

In [2]:
path = '../data/BTCdata_presim.csv'
df = pd.read_csv(path)
df['time'] = pd.to_datetime(df['time'])
df = df.rename(columns={'blocktime': 'block_times', 'price': 'prices', 'probability of success /Eh': 'winning_rates'})
#df.head()

In [3]:
# test with small number of iter
generate_simulation_data_asert(num_iter=2, prev_data=df)

In [5]:
# takes 2 hours
generate_simulation_data_asert(num_iter=5000, prev_data=df)

In [2]:
dir_sim = '/Users/kyo/prog/DAA_Lucas/simulation/'
df_asert_winrate = pd.read_csv(dir_sim + 'DAA_asert_winrate_ps0_5000obs_T=None.csv', index_col=0)
df_asert_optwinrate = pd.read_csv(dir_sim + 'DAA_asert_optwinrate_ps0_5000obs_T=None.csv', index_col=0)

In [3]:
error_asert = MSE(df1=df_asert_winrate, df2=df_asert_optwinrate)
error_asert

2.0209362721203346e-12

In [15]:
error_DAA2 = 5.6909088461042424e-12

In [16]:
error_DAA2/error_asert

2.81597639896553

In [17]:
error_DAA1 = 1.36e-10
error_DAA1/error_asert

67.29554111932036

## MSE Comparison
* MSE of ASERT is $2.02 \times 10^{-12}$.
* MSE of DAA-2 is $5.69 \times 10^{-12}$.
* MSE of DAA-1 is $1.36 \times 10^{-10}$.

* ASERT is the best in terms of MSE among these three
* The order is the same as DAA-2, but still it has less MSE.

## Blocktime Statistics

In [3]:
dir_sim = '/Volumes/Data/research/BDA/simulation/'
filelist = [dir_sim+'DAA_asert_blocktime_ps0_5000obs_T=None.csv']
df_stats = make_stats(filelist=filelist, dir_sim=dir_sim)

In [4]:
df_stats

DAA_asert_blocktime_ps0_5000obs_T=None.csv
mean                                      11.267700
std                                       10.821421
over60                                     1.000000
over120                                    0.114377
over180                                    0.000600